File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    d = [0] * 100001
    
    d[0] = 0
    d[1] = 1
    for i in range(2, 100001) :
        d[i] = d[i - 2] + d[i - 1]
        
    return d[n] % 1234567

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        
        long[] temp = new long[100001];
        temp[1] = 1;
        for (int i = 2; i < temp.length; i++) {
            temp[i] = (temp[i - 1] + temp[i - 2]) % 1234567;
        }
        
        return (int) temp[n];
    }
}